common/assertion-functions/src/main/kotlin/com/testerum/common_assertion_functions/functions/builtin/ArrayFunctions.kt

```diff
@@ -0,0 +1,108 @@
+package com.testerum.common_assertion_functions.functions.builtin
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.NullNode
+import com.testerum.common_assertion_functions.functions.AssertionFailedException
+import com.testerum.common_assertion_functions.functions.AssertionFunction
+import java.math.BigDecimal
+
+object ArrayFunctions {
+
+    @AssertionFunction
+    fun isArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+    
+    @AssertionFunction
+    fun isArrayOrNull(actualNode: JsonNode) {
+        if (actualNode is NullNode) {
+            return
+        }
+
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isNonEmptyArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            val arrayNode = actualNode.get(0)
+            if (arrayNode != null) {
+                val actualSize = arrayNode.size()
+                if (actualSize > 0) {
+                    return
+                } else {
+                    throw AssertionFailedException("expected an non empty Array, but got an Array with ${actualSize} ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
```

Comment:
```suggestion
                    throw AssertionFailedException("expected a non empty Array, but got an Array with ${actualSize} ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
```
The rule for ``a`` or ``an`` is:
* use ``a`` if the following word begins with a vowel-like sound (e.g. ``a child``, ``a line``)
* use ``an`` if the following word begins with a consonant-like sound (e.g. ``an arm", "an umbrella")

Note that it's the sound that determines ``a`` or ``an``, not the letters.
For example the Americans would say ``an herb`` (because they don't pronounce the ``h``), while the British would say ``a herb``). This doesn't happen often, so you shouldn't worry too much about it.
Just remember that you use ``a`` or ``an`` depending if the first syllable of the next word is a vowel-sound or a consonant-sound.